roborazzi-accessibility-check/src/main/java/com/github/takahirom/roborazzi/RoborazziATFAccessibilityChecker.kt

```diff
@@ -273,3 +275,21 @@ data class AccessibilityCheckAfterTestStrategy(
       )
   }
 }
+
+@ExperimentalRoborazziApi
+data class AccessibilityCheckEachScreenshotStrategy(
+  override val accessibilityOptionsFactory: () -> RoborazziATFAccessibilityCheckOptions = { provideATFAccessibilityOptionsOrCreateDefault() },
+) : BaseAccessibilityCheckStrategy() {
+  override fun shouldRunAt(checkPoint: CheckPoint): Boolean {
```

Comment:
You might already be thinking this, but how about using interface functions directly?  

```kotlin
// AccessibilityCheckStrategy.kt
interface AccessibilityCheckStrategy {
    fun afterScreenshot(captureRoot: CaptureRoot, roborazziOptions: RoborazziOptions) {}
    fun afterTest(captureRoot: CaptureRoot, roborazziOptions: RoborazziOptions) {}
}

// BaseAccessibilityCheckStrategy.kt
abstract class BaseAccessibilityCheckStrategy : AccessibilityCheckStrategy {
    abstract val accessibilityOptionsFactory: () -> RoborazziATFAccessibilityCheckOptions

    private val accessibilityOptions by lazy { accessibilityOptionsFactory() }

    protected fun runChecks(captureRoot: CaptureRoot, roborazziOptions: RoborazziOptions) {
        accessibilityOptions
            .checker
            .runAccessibilityChecks(
                captureRoot = captureRoot,
                roborazziOptions = roborazziOptions,
                options = accessibilityOptions
            )
    }
}

// AccessibilityCheckEachScreenshotStrategy.kt
data class AccessibilityCheckEachScreenshotStrategy(
    override val accessibilityOptionsFactory: () -> RoborazziATFAccessibilityCheckOptions = { provideATFAccessibilityOptionsOrCreateDefault() },
) : BaseAccessibilityCheckStrategy() {
    override fun afterScreenshot(captureRoot: CaptureRoot, roborazziOptions: RoborazziOptions) {
        runChecks(captureRoot, roborazziOptions)
    }

    override fun afterTest(captureRoot: CaptureRoot, roborazziOptions: RoborazziOptions) {
        runChecks(captureRoot, roborazziOptions)
    }
}

// AccessibilityCheckAfterTestStrategy.kt
data class AccessibilityCheckAfterTestStrategy(
    override val accessibilityOptionsFactory: () -> RoborazziATFAccessibilityCheckOptions = { provideATFAccessibilityOptionsOrCreateDefault() },
) : BaseAccessibilityCheckStrategy() {
    override fun afterTest(captureRoot: CaptureRoot, roborazziOptions: RoborazziOptions) {
        runChecks(captureRoot, roborazziOptions)
    }
}

// RoborazziRule.kt modification
class RoborazziRule private constructor(
    // ... omitted ...
) {
    // ... omitted ...

    // Directly call the event-specific method when needed
    accessibilityChecks.afterScreenshot(captureRoot, options.roborazziOptions)

    // ... omitted ...

    // Also call the method at the end of the test
    accessibilityChecks.afterTest(captureRoot, options.roborazziOptions)
}
```

Reply:
Good idea.